packages/ui/src/providers/delete-modal.provider.test.tsx

```diff
@@ -0,0 +1,52 @@
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import { useContext } from 'react';
+import { DeleteModalContextProvider, DeleteModalContext } from './delete-modal.provider';
+
+let deleteConfirmationResult: boolean | undefined;
+
+describe('DeleteModalProvider', () => {
+  it('calls deleteConfirmation with true when Confirm button is clicked', async () => {
+    render(
+      <DeleteModalContextProvider>
+        <TestComponent />
+      </DeleteModalContextProvider>,
+    );
+
+    const deleteButton = screen.getByText('Delete');
+    fireEvent.click(deleteButton);
+
+    const confirmButton = await screen.findByRole('button', { name: 'Confirm' });
+    fireEvent.click(confirmButton);
+
+    // Wait for deleteConfirmation promise to resolve
+    await waitFor(() => expect(deleteConfirmationResult).toEqual(true));
```

Comment:
I think in this case we need to make the inner function `async`

Reply:
I stand corrected, according to [the docs](https://testing-library.com/docs/dom-testing-library/api-async/)

>If you return a promise in the waitFor callback (either explicitly or implicitly with the async syntax), then the waitFor utility does not call your callback again until that promise rejects. This allows you to waitFor things that must be checked asynchronously.

No change is required and sorry for the confusion.